util/src/test/java/tech/pegasys/pantheon/util/FutureUtilsTest.java

```diff
@@ -188,6 +192,50 @@ public void shouldNotCallErrorHandlerWhenFutureCompletesSuccessfully() {
     assertThat(result).isCompletedWithValue("Done");
   }
 
+  @Test
+  public void shouldReturnMatchingSuccessfulFutureWhenUsingAnyOfMatching() {
+    final CompletableFuture<String> future1 = new CompletableFuture<>();
+    final CompletableFuture<String> future2 = new CompletableFuture<>();
+    final CompletableFuture<String> future3 = new CompletableFuture<>();
+    future1.complete("a");
```

Comment:
it it worth showing that the future can complete *after* the creation of the "collation" future? (rather than completing them all prior to construction)?
Or maybe show that it is not complete if a sub-item completes with an non-matching value? I.e. do a multi-step test?

Reply:
I think it is, will try add something